app/models/concerns/taxonomix.rb

```diff
@@ -19,7 +19,7 @@ module Taxonomix
     scoped_search :relation => :organizations, :on => :id, :rename => :organization_id, :complete_enabled => false, :only_explicit => true, :validator => ScopedSearch::Validators::INTEGER
 
     dirty_has_many_associations :organizations, :locations
-    audit_associations :organizations, :locations
+    audit_associations :organizations, :locations if self.respond_to? :audit_associations
```

Comment:
We include `Taxonomix` in `Audited::Audit`, which inherits from `ActiveRecord` and not `ApplicationRecord`, so it doesn't respond to `audit_associations` any longer. This is intentional, we only want models inheriting `ApplicationRecord` (i.e. those defined by us) to be able to audit their associations.

Reply:
right, so `Audited::Audit` should start inheriting from ApplicationRecord at some point? and then the condition is not required?

Reply:
Even if it would be Rails 5 compatible only? ApplicationRecord is common ancestor for all models by default so I don't understand why audited gem would extend core rails class. But that's definitely unrelated to this PR. I just don't like the fact the definition would be skipped silently. Would it be possible to log it? If this applies only for Audited::Audit, it would be one log on app start. If that happens for any other resource, something is wrong and it's better to know.

Reply:
No, Audited::Audit will always inherit ActiveRecord, since we want it to work as expected for all rails applications, even if they have some wierd monkey-patching in place in their ApplicationRecords that may break the gem. See https://blog.bigbinary.com/2015/12/28/application-record-in-rails-5.html for more info regarding this change.

Reply:
From that linked article

> Now ApplicationRecord will be a single point of entry for all the customizations and extensions needed for an application, instead of monkey patching ActiveRecord::Base.

Isn't auditing doing exactly this? Or do you see that as an extension for ActiveRecord::Base itself? That would explain our different views on this. Anyway, I'm fine with this but as I said, debug log in case the method is not available would be good I think.

Reply:
I had two thoughts about it. I would like to share.

1. About adding extra wrapper around `Taxonomix`. It won't fit into our scenario as we have already used `Taxonomix`  module at many places.
2. I know that it is not an appropriate way but would it be fixed by using `remove_method`?
I mean instead of using `Taxonomix` module, we use copy it using ` .dup` & then remove that method.

Again,  `respond_to?` is small one line change to fix this instead of adding extra wrapper or using `.dup` and `remove_method`.

Reply:
So I think `respond_to?` is a go.